keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -366,6 +365,12 @@ class SqlDeliveryConfigRepository(
             .set(ENVIRONMENT_RESOURCE.RESOURCE_UID, select(RESOURCE.UID).from(RESOURCE).where(RESOURCE.ID.eq(resource.id)))
             .onDuplicateKeyIgnore()
             .execute()
+          // delete any other environment's link to this same resource (in case we are moving it
+          // from one environment to another)
+          jooq.deleteFrom(ENVIRONMENT_RESOURCE)
```

Comment:
Shouldn't this be in the scope of a transaction along with the other statements?

Reply:
That entire function seems to be non-transactional for some reason. I vaguely remember there was some thinking behind that but I can't recall what.

Reply:
Okay, I figured out why. The only call path to this method is from `CombinedRepository.upsertDeliveryConfig` which uses Spring's annotation-bounded transactions.

Reply:
Right, but if I recall correctly, that does not automagically propagate to jooq. I may be wrong.

Reply:
The entire point of `DeliveryConfigTransactionTests` is to prove that it does